packages/flutter/lib/src/gestures/events.dart

```diff
@@ -21,18 +21,66 @@ enum PointerDeviceKind {
   mouse
 }
 
+/// The bit of [PointerEvent.buttons] that coorresponds to the primary mouse button.
+///
+/// The primary mouse button is typically the left button on the top of the
+/// mouse but can be reconfigured to be a different physical button.
 const int kPrimaryMouseButton = 0x01;
+
+/// The bit of [PointerEvent.buttons] that coorresponds to the secondary mouse button.
+///
+/// The secondary mouse button is typically the right button on the top of the
+/// mouse but can be reconfigured to be a different physical button.
 const int kSecondaryMouseButton = 0x02;
+
+/// The bit of [PointerEvent.buttons] that coorresponds to the primary stylus button.
+///
+/// The primary stylus button is typically the top of the stylus and near the
+/// tip but can be reconfigured to be a different physical button.
 const int kPrimaryStylusButton = 0x02;
+
+/// The bit of [PointerEvent.buttons] that coorresponds to the middle mouse button.
+///
+/// The middle mouse button is typically between the left and right buttons on
+/// the top of the mouse but can be reconfigured to be a different physical
+/// button.
 const int kMiddleMouseButton = 0x04;
+
+/// The bit of [PointerEvent.buttons] that coorresponds to the secondary stylus button.
+///
+/// The secondary stylus button is typically on the end of the stylus fartherest
+/// from the tip but can be reconfigured to be a different physical button.
 const int kSecondaryStylusButton = 0x04;
+
+/// The bit of [PointerEvent.buttons] that coorresponds to the back mouse button.
+///
+/// The back mouse button is typically on the left side of the mouse but can be
+/// reconfigured to be a different physical button.
 const int kBackMouseButton = 0x08;
-const int forwardMouseButton = 0x10;
+
+/// The bit of [PointerEvent.buttons] that coorresponds to the forward mouse button.
+///
+/// The forward mouse button is typically on the right side of the mouse but can
+/// be reconfigured to be a different physical button.
+const int kForwardMouseButton = 0x10;
+
+/// The bit of [PointerEvent.buttons] that coorresponds to the nth mouse button.
+///
+/// See [kPrimaryMouseButton], [kSecondaryMouseButton], [kMiddleMouseButton],
+/// [kBackMouseButton], and [kForwardMouseButton] for semantic names for some
+/// mouse buttons.
 int nthMouseButton(int number) => kPrimaryMouseButton << (number - 1);
```

Comment:
turns out that for efficiency we should do `(kPrimaryMouseButton << (number - 1)) & 0x3FFFFFFFFFFFFFFF` to ensure that the value remains a smi. See https://www.dartlang.org/articles/numeric-computation/#smis-and-mints

(Except that should probably be a constant in foundation/basic_types.)